OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -421,4 +421,10 @@ public interface ITargetableCells
 
 	[RequireExplicitImplementation]
 	public interface IPreventsShroudReset { bool PreventShroudReset(Actor self); }
+
+	[RequireExplicitImplementation]
+	public interface IResourcePurifier
```

Comment:
Can we rename this to `IResourceValueModifier` and the method to `int GetResourceValueModifier()` to be consistent with our other modifier interfaces?

Reply:
https://github.com/ABrandau/OpenRA/blob/Sequence_Exception_Fix/OpenRA.Mods.AS/Traits/ResourcePurifier.cs currently adds that cash bonus separately, changing the method to the above would require a larger rewrite of the downstream trait (which currently also displays the additional cash over the modifying actor).

I'd prefer to change this into an `INotifyRefineryCashGiven` and change the method to `OnCashGiven(int amount)` instead, then we can stick to renaming for now.

I don't really feel like starting to refactor downstream traits on top of our own...

Reply:
This is an interface for my trait tbh and I'd be more than okay with submitting the whole trait entirely instead of keeping the interface alone, I just had no time to do this (this is the RA2 Purifier logic afterall).

Reply:
Also, the separate bonus is intended, because that's how RA2 works.

Reply:
So yes, if we can decide on the namings, I guess I can take the blame and able to submit this with suitable names during the weekend.

Reply:
I have edited Purifier logic that allows showing cash tick on the refinery rather than purifier also filtering per refinery or harvester. But latter requires passing the harvester's name a lot not sure if it is good. I use that for Generals Worker Shoes and Supply Lines upgrades.

Reply:
Split this off to #15283.